db/system_keyspace.cc

```diff
@@ -1015,6 +1018,28 @@ schema_ptr system_keyspace::broadcast_kv_store() {
     return schema;
 }
 
+schema_ptr system_keyspace::tablets() {
+    static thread_local auto schema = [] {
+        // FIXME: Allow UDTs in system keyspace:
+        // CREATE TYPE tablet_replica (replica_id uuid, shard int);
+        // replica_set_type = frozen<set<tablet_replica>>
+        auto replica_set_type = set_type_impl::get_instance(tuple_type_impl::get_instance({uuid_type, int32_type}), false);
+        auto id = generate_legacy_id(NAME, TABLETS);
+        return schema_builder(NAME, TABLETS, id)
+            .with_column("keyspace_name", utf8_type, column_kind::partition_key)
+            .with_column("table_id", uuid_type, column_kind::partition_key)
+            .with_column("tablet_count", int32_type, column_kind::static_column)
+            .with_column("table_name", utf8_type, column_kind::static_column)
+            .with_column("last_token", long_type, column_kind::clustering_key)
+            .with_column("replicas", replica_set_type)
+            .with_column("new_replicas", replica_set_type)
+            .with_column("stage", utf8_type)
+            .with_version(generate_schema_version(id))
+            .build();
+    }();
+    return schema;
+}
+
```

Comment:
Is this not in all_tables()?  I don't think it's good to create it on-demand.

Reply:
It follows the same pattern as other tables, the schema is kept in thread_local so it's initialied once.